mlir/test/Dialect/LLVMIR/constant-folding.mlir

```diff
@@ -101,3 +101,63 @@ llvm.func @addressof_blocks(%arg: i1) -> !llvm.ptr {
 }
 
 llvm.mlir.global constant @foo() : i32
+
+// -----
+
+// CHECK-LABEL: llvm.func @zero
+llvm.func @zero() {
+  // CHECK-NEXT: %[[ZERO:.+]] = llvm.mlir.zero : i32
```

Comment:
Interesting I though this would be a llvm.constant 0 : i32

Reply:
I've only seen `llvm.mlir.zero` used for `nullptr`, what's the convention here?

Reply:
It can in principle be used everywhere. I think we want to fold it to integer / float zero attributes if possible but we want to keep it for pointers / structs / arrays where there is no good alternative.